core/lib/phases.js

```diff
@@ -94,83 +94,46 @@ function createPause(spec, ee) {
 
 function createRamp(spec, ee) {
   const duration = spec.duration || 1;
-  const arrivalRate = spec.arrivalRate || 0;
-  const rampTo = spec.rampTo || arrivalRate + 1;
+  const arrivalRate = spec.arrivalRate;
+  const rampTo = spec.rampTo;
 
-  const tick = 1000 / Math.max(arrivalRate, rampTo); // smallest tick
-  const r0 = arrivalRate; // initial arrival rate
+  const tick = Math.min(1000 / Math.max(arrivalRate, rampTo), 1000); // smallest tick, capped at 500ms
   const difference = rampTo - arrivalRate;
-  const offset = difference < 0 ? -1 : 1;
-  const periods = Math.abs(difference) + 1;
-  const ticksPerPeriod = ((duration / periods) * 1000) / tick;
-  const periodLenSec = duration / periods;
-
-  let expected = 0;
-  for (let i = 1; i <= periods; i++) {
-    let expectedInPeriod = periodLenSec * i;
-    expected += expectedInPeriod;
-  }
-  expected = Math.floor(expected);
-
-  // console.log(`expecting ${expected} total arrivals`);
-
-  let probabilities = crypto.randomBytes(
-    Math.ceil(((duration * 1000) / tick) * 1.25)
-  );
+  const periods = duration * 1000 / tick;
+
+  function arrivalProbability(currentStep) {
+    // linear function ax + b
+    // normalized to 0 <= f(t) <= 1
+    // Anything under function value should be an arrival
+
+    let t = currentStep * tick / 1000;
+    return ((difference / duration) * t + arrivalRate) / Math.max(difference, arrivalRate);
+  };
+
+  let probabilities = Array.from({length: periods}, () => Math.random());
```

Comment:
Wasn't really sure we need a cryptographically secure generation here. Using `Math.random` should be more performant